src/xo-app/new/sr/index.js

```diff
@@ -0,0 +1,620 @@
+import _, { messages } from 'messages'
+import ActionButton from 'action-button'
+import filter from 'lodash/filter'
+import Icon from 'icon'
+import includes from 'lodash/includes'
+import info, { error } from 'notification'
+import isEmpty from 'lodash/isEmpty'
+import map from 'lodash/map'
+import React, { Component } from 'react'
+import trim from 'lodash.trim'
+import Wizard, { Section } from 'wizard'
+import { confirm } from 'modal'
+import { connectStore, formatSize } from 'utils'
+import { GenericSelect, SelectHost } from 'select-objects'
+import { hosts, objects, createFilter, createSelector } from 'selectors'
+import { injectIntl } from 'react-intl'
+
+class SelectIqn extends GenericSelect {
+  _computeOptions (props) {
+    return map(props.options, iqn => ({
+      value: iqn,
+      label: `${iqn.iqn} (${iqn.ip})`
+    }))
+  }
+}
+
+class SelectLun extends GenericSelect {
+  _computeOptions (props) {
+    return map(props.options, lun => ({
+      value: lun,
+      label: `LUN ${lun.id}: ${lun.serial} - ${formatSize(+lun.size)} - (${lun.vendor})`
+    }))
+  }
+}
+
+import {
+  createSrIso,
+  createSrIscsi,
+  createSrLvm,
+  createSrNfs,
+  probeSrIscsiExists,
+  probeSrIscsiIqns,
+  probeSrIscsiLuns,
+  probeSrNfs,
+  probeSrNfsExists,
+  reattachSrIso,
+  reattachSr
+} from 'xo'
+
+// ===================================================================
+
+const SR_TYPE_TO_INFO = {
+  nfs: 'NFS',
+  iscsi: 'iSCSI',
+  lvm: 'Local LVM',
+  local: 'Local',
+  nfsiso: 'NFS ISO',
+  smb: 'SMB'
+}
+
+// ===================================================================
+
+@injectIntl
+@connectStore(() => {
+  const srs = createFilter(objects, () => object => object.type === 'SR', true)
+
+  return (state, props) => ({
+    hosts: hosts(state, props),
+    srs: srs(state, props)
+  })
+})
+export default class New extends Component {
+  static contextTypes = {
+    router: React.PropTypes.object
+  }
+
+  constructor (props) {
+    super(props)
+    this.state = {
+      description: undefined,
+      host: undefined,
+      iqn: undefined,
+      iqns: undefined,
+      lockCreation: undefined,
+      lun: undefined,
+      luns: undefined,
+      name: undefined,
+      path: undefined,
+      paths: undefined,
+      type: undefined,
+      unused: undefined,
+      usage: undefined,
+      used: undefined
+    }
+    this.getHostSrs = createFilter(
+      () => this.props.srs,
+      createSelector(
+        () => this.state.host,
+        ({$pool, id}) => sr => sr.$container === $pool || sr.$container === id
+      ),
+      true
+    )
+  }
+
+  _handleSubmit = async () => {
+    const {
+      device,
+      localPath,
+      name,
+      description,
+      server,
+      username,
+      password
+    } = this.refs
+    const {
+      host,
+      iqn,
+      lun,
+      path,
+      type
+    } = this.state
+    const [address, port] = server && server.value.split(':') || []
```

Comment:
What is this parsing?
We must avoid parsing stuff everywhere, it should only be done in dedicated modules/functions.

Also, parsing is difficult, a brute split like this one might would break IPv6.